zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -216,54 +223,68 @@ CREATE TABLE orchard_received_note_spends (
-CREATE TABLE "utxos" (
+///
+/// ### Columns:
+/// - `id`: Primary key
+/// - `transaction_id`: Reference to the transaction in which this TXO was created
+/// - `output_index`: The output index of this TXO in the transaction referred to by `transaction_id`
+/// - `account_id`: The account that controls spend authority for this TXO
+/// - `address`: The address to which this TXO was sent
+/// - `script`: The full txout script
+/// - `value_zat`: The value of the TXO in zatoshis
+/// - `max_observed_unspent_height`: The maximum block height at which this TXO was observed
+///   to be a member of the UTXO set. This is intended to be used to determine when the TXO
+///   is no longer a part of the UTXO set, in the case that the transaction that spends it is
+///   not detected by the wallet.
+pub(super) const TABLE_TRANSPARENT_RECEIVED_OUTPUTS: &str = r#"
+CREATE TABLE transparent_received_outputs (
     id INTEGER PRIMARY KEY,
-    received_by_account_id INTEGER NOT NULL,
+    transaction_id INTEGER NOT NULL,
+    output_index INTEGER NOT NULL,
+    account_id INTEGER NOT NULL,
     address TEXT NOT NULL,
-    prevout_txid BLOB NOT NULL,
-    prevout_idx INTEGER NOT NULL,
     script BLOB NOT NULL,
     value_zat INTEGER NOT NULL,
-    height INTEGER NOT NULL,
-    FOREIGN KEY (received_by_account_id) REFERENCES accounts(id),
-    CONSTRAINT tx_outpoint UNIQUE (prevout_txid, prevout_idx)
+    max_observed_unspent_height INTEGER,
+    FOREIGN KEY (transaction_id) REFERENCES transactions(id_tx),
+    FOREIGN KEY (account_id) REFERENCES accounts(id),
+    CONSTRAINT transparent_output_unique UNIQUE (transaction_id, output_index)            
 )"#;
-pub(super) const INDEX_UTXOS_RECEIVED_BY_ACCOUNT: &str =
-    r#"CREATE INDEX utxos_received_by_account ON "utxos" (received_by_account_id)"#;
+pub(super) const INDEX_TRANSPARENT_RECEIVED_OUTPUTS_ACCOUNT_ID: &str = r#"
+CREATE INDEX idx_transparent_received_outputs_account_id 
+ON "transparent_received_outputs" (account_id)"#;
 
-/// A junction table between received transparent outputs and the transactions that spend
-/// them.
+/// A junction table between received transparent outputs and the transactions that spend them.
 ///
-/// This is identical to [`TABLE_SAPLING_RECEIVED_NOTE_SPENDS`]; see its documentation for
-/// details. Note however that [`TABLE_UTXOS`] and [`TABLE_SAPLING_RECEIVED_NOTES`] are
-/// not equivalent, and care must be taken when interpreting the result of joining this
-/// table to [`TABLE_UTXOS`].
-pub(super) const TABLE_TRANSPARENT_RECEIVED_OUTPUT_SPENDS: &str = "
-CREATE TABLE transparent_received_output_spends (
+/// This is identical to [`TABLE_SAPLING_RECEIVED_NOTE_SPENDS`]; see its documentation for details.
+/// Note however that [`TABLE_TRANSPARENT_RECEIVED_OUTPUTS`] and [`TABLE_SAPLING_RECEIVED_NOTES`]
+/// are not equivalent, and care must be taken when interpreting the result of joining this table
+/// to [`TABLE_TRANSPARENT_RECEIVED_OUTPUTS`].
```

Comment:
This comment just left me anxious that I was missing something, without increasing my understanding. :sweat_smile:

```suggestion
/// This plays the same role for transparent TXOs as does [`TABLE_SAPLING_RECEIVED_NOTE_SPENDS`]
/// for Sapling notes. However, [`TABLE_TRANSPARENT_RECEIVED_OUTPUTS`] and [`TABLE_SAPLING_RECEIVED_NOTES`] differ in that [fill in here precisely *how* they differ, and *what* care must be taken in joining them].
```